lib/AuthStateManager.ts

```diff
@@ -92,12 +92,22 @@ export class AuthStateManager {
       devMode && log('emitted');
     };
 
+    const finalPromise = (origPromise) => {       
```

Comment:
How about making the finalPromise returns a function instead of a promise? In this way, I think we can resolve the concern about the promise may become stale.